core/client_test.go

```diff
@@ -0,0 +1,52 @@
+package core
+
+import (
+	"context"
+	"testing"
+
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestClientLifecycle(t *testing.T) {
+	client := MockClient()
+	err := client.Stop()
+	require.NoError(t, err)
+}
+
+func TestClient_Status(t *testing.T) {
+	client := MockClient()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	status, err := client.Status(ctx)
+	require.NoError(t, err)
+	assert.NotNil(t, status)
+
+	err = client.Stop()
+	require.NoError(t, err)
+}
+
+func TestClient_StartBlockSubscription_And_GetBlock(t *testing.T) {
+	client := MockClient()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(cancel)
+
+	eventChan, err := client.Subscribe(ctx, "NewBlock/Events", types.QueryForEvent(types.EventNewBlock).String())
```

Comment:
should we extract `"NewBlock/Events"` into a local const?

Reply:
Yes pls.

Reply:
Should be resolved by https://github.com/celestiaorg/celestia-node/pull/55